tests/test_components.py

```diff
@@ -538,114 +543,128 @@ def test_parse(self):
             workflows={"wf1": self.wf1},
-            pages={"page1": self.page1, "page2": self.page2},
+            pages=pages
         )
-        self.component = Collections(self.bootstrap)
-
-    def test_parse_generates_collection(self):
-        """
-        Test that parsing is able to call the Collection.get_or_create method
-        """
-        component = self.component
+        component = Collections(bootstrap)
         component.raw_data = {
             "collection1": {
-                "pages": ["page1", "page2"],
+                "pages": pages.keys(),
                 "name": "Collection 1",
                 "user": "user1",
-                "workflow": "wf1",
+                "workflow": "wf1"
             }
         }
+        return component
 
-        with patch.object(component, "get_or_create") as get_or_create:
-            component.parse()
-            get_or_create.assert_called_once_with(
-                {"workflow": self.wf1, "name": "Collection 1", "author": self.user}
-            )
+    @freeze_time()
+    def test_parse_creates_collection_in_db(self):
+        """If the ModerationCollection doesn't exist, a new instance
+        should be created in the db"""
+        component = self._get_collections_obj()
 
-    def test_parse_stores_collection(self):
-        """
-        Test that parsing is able to call the Collection.get_or_create method
-        """
-        component = self.component
-        component.raw_data = {
-            "collection1": {
-                "pages": ["page1"],
-                "name": "Collection 1",
-                "user": "user1",
-                "workflow": "wf1",
-            }
-        }
-        with patch.object(component, "get_or_create") as get_or_create:
-            component.parse()
-            collection_data = {
-                "author": self.user,
-                "workflow": self.wf1,
-                "name": "Collection 1",
-            }
-            self.assertEqual(
-                component.data["collection1"], get_or_create(collection_data)
-            )
+        component.parse()
 
-    def test_get_or_create(self):
-        # test add
-        collection_data = {
-            "author": self.user,
-            "workflow": self.wf1,
-            "name": "Collection 1",
-        }
-        collection = self.component.get_or_create(collection_data)
-        self.assertEqual(collection.id, 1)
-        collection_data2 = {
-            "author": self.user,
-            "workflow": self.wf1,
-            "name": "Collection 2",
-        }
-        collection2 = self.component.get_or_create(collection_data2)
-        self.assertEqual(collection2.id, 2)
+        collection = ModerationCollection.objects.get()
+        self.assertEqual(collection.name, "Collection 1")
+        self.assertEqual(collection.author, self.user)
+        self.assertEqual(collection.workflow, self.wf1)
+        # TODO: Is it correct that all collections created with json
```

Comment:
Probably a question to @geoidesic

Reply:
I'm not sure I understand the question... but I'm going with "yes".